test/test_tensorflow.py

```diff
@@ -646,35 +924,52 @@ def test_compression_fp16(self):
+        def graph_and_eager(session=None):
+            for dtype in valid_dtypes:
+                tensor = tf.ones(tensor_size, dtype=dtype)
 
-            tensor_compressed, ctx = compression.compress(tensor)
-            self.assertEqual(tensor_compressed.dtype, tf.float16)
+                tensor_compressed, ctx = compression.compress(tensor)
+                self.assertEqual(tensor_compressed.dtype, tf.float16)
 
-            tensor_decompressed = compression.decompress(tensor_compressed, ctx)
-            self.assertEqual(tensor_decompressed.dtype, dtype)
+                tensor_decompressed = compression.decompress(
+                    tensor_compressed, ctx)
+                self.assertEqual(tensor_decompressed.dtype, dtype)
 
-            actual = self.evaluate(tensor_decompressed)
-            expected = np.ones(tensor_size)
-            err = np.linalg.norm(expected - actual)
-            self.assertLess(err, 0.00000001)
+                if session:
+                    actual = session.run(tensor_decompressed)
+                else:
+                    actual = tensor_decompressed
+                expected = np.ones(tensor_size)
+                err = np.linalg.norm(expected - actual)
+                self.assertLess(err, 0.00000001)
+
+            for dtype in invalid_dtypes:
+                if not session and dtype is tf.bool:
+                    return
+                tensor = tf.ones(tensor_size, dtype=dtype)
 
-        for dtype in invalid_dtypes:
-            tensor = tf.ones(tensor_size, dtype=dtype)
+                tensor_compressed, ctx = compression.compress(tensor)
+                self.assertEqual(tensor_compressed.dtype, dtype)
 
-            tensor_compressed, ctx = compression.compress(tensor)
-            self.assertEqual(tensor_compressed.dtype, dtype)
+                tensor_decompressed = compression.decompress(
+                    tensor_compressed, ctx)
+                self.assertEqual(tensor_decompressed.dtype, dtype)
+
+                if session:
+                    actual = session.run(tensor_decompressed)
+                else:
+                    actual = tensor_decompressed
+                expected = np.ones(tensor_size)
+                err = np.linalg.norm(expected - actual)
+                self.assertLess(err, 0.00000001)
 
-            tensor_decompressed = compression.decompress(tensor_compressed, ctx)
-            self.assertEqual(tensor_decompressed.dtype, dtype)
+        if _executing_eagerly():
+            with self.eager_mode:
+                graph_and_eager()
 
-            actual = self.evaluate(tensor_decompressed)
-            expected = np.ones(tensor_size)
-            err = np.linalg.norm(expected - actual)
-            self.assertLess(err, 0.00000001)
+        with self.test_session(config=self.config) as session:
+            graph_and_eager(session)
 
 
 if __name__ == '__main__':
-    run_all_in_graph_and_eager_modes(MPITests)
```

Comment:
I wonder... could we have a single IF here to test if eager mode is available, and then use it, w/o changing rest of the code?